### DISCUSSION TITLE
现在partition_tag参数是不是已经不使用了？

### DISCUSSION THREAD

**Discussion by turbochen2018** [milvus-io/milvus#16544] [Q&A and General discussion]
原来可以通过年月日组合名字来创建partition，然后可以通过partition_tag的正则表达式来过滤时间删选（如下链接文章所描述的一样）那现在如何较好的实现时间筛选过滤？
https://zhuanlan.zhihu.com/p/91295342   Milvus 0.6.0新增功能：分区表

**Comment by yhmo** [comment]
在2.0里每一条数据都有一个时间戳，search()方法有一个travel_timestamp参数，可以用这个参数设置一个时间点，在这个时间点以前的数据会被搜索到，这个时间点之后的数据不会被搜索到。
https://milvus.io/docs/v2.0.x/timetravel.md

**Comment by yhmo** [comment]
另外一个做法是：定义一个int64字段，用来存放你自己定义的时间戳。在insert时把当时的时间戳一起录入。在调用search()的时候有个“expr”参数，用来输入一个过滤表达式。假设你定义的int64字段叫做“ts”，那么在搜索的时候如果你想搜索某个时间区间的数据，就可以这样调用：search(..., expr="1000 < ts < 2000")

**Reply by xiaofan-luan** [reply]
> 原来可以通过年月日组合名字来创建partition，然后可以通过partition_tag的正则表达式来过滤时间删选（如下链接文章所描述的一样）那现在如何较好的实现时间筛选过滤？ https://zhuanlan.zhihu.com/p/91295342 Milvus 0.6.0新增功能：分区表

Milvus 2.0 也有partition功能，现在的问题是因为缺少正则匹配能力么？

**Reply by yhmo** [reply]
> > 原来可以通过年月日组合名字来创建partition，然后可以通过partition_tag的正则表达式来过滤时间删选（如下链接文章所描述的一样）那现在如何较好的实现时间筛选过滤？ https://zhuanlan.zhihu.com/p/91295342 Milvus 0.6.0新增功能：分区表
> 
> Milvus 2.0 也有partition功能，现在的问题是因为缺少正则匹配能力么？

v1.0没有过滤数据的能力，所以允许用户用partition的名字来做正则匹配做粗略的过滤。v2.0能通过表达式过滤数据，不需要这么做。
